bitcoin/src/blockdata/block.rs

```diff
@@ -311,8 +317,8 @@ impl BlockCheckedExt for Block<Checked> {
             return Err(Bip34Error::Unsupported);
         }
 
-        let cb = self.coinbase().ok_or(Bip34Error::NotPresent)?;
-        let input = cb.input.first().ok_or(Bip34Error::NotPresent)?;
+        let cb = self.coinbase();
+        let input = cb.as_transaction().input.first().ok_or(Bip34Error::NotPresent)?;
```

Comment:
This needs to be infallible since a valid coinbase is guaranteed to have at least one input.